examples/kubernetes-guestbook/simple/Main.scala

```diff
@@ -0,0 +1,166 @@
+          metadata = ObjectMetaArgs(labels = redisReplicaLabels),
+          spec = PodSpecArgs(containers =
+            List(
+              ContainerArgs(
+                name = "replica",
+                image = "pulumi/guestbook-redis-replica",
+                resources = ResourceRequirementsArgs(
+                  requests = Map(
+                    "cpu" -> "100m",
+                    "memory" -> "100Mi"
+                  )
+                ),
+                // If your cluster config does not include a dns service, then to instead access an environment
+                // variable to find the leader's host, change `value: "dns"` to read `value: "env"`.
+                env = List(EnvVarArgs("name", "GET_HOSTS_FROM"), EnvVarArgs("value", "dns")),
+                ports = List(ContainerPortArgs(containerPort = 6379))
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+
+  val redisReplicaService = Service(
+    "redis-replica",
+    ServiceArgs(
+      metadata = ObjectMetaArgs(labels = redisReplicaLabels, name = "redis-replica"),
+      spec = ServiceSpecArgs(
+        ports = List(ServicePortArgs(port = 6379, targetPort = 6379)),
+        selector = redisReplicaLabels
+      )
+    ),
+    opts = opts(dependsOn = redisReplicaDeployment)
+  )
+
+  //
+  // FRONTEND
+  //
+  val frontendLabels = Map("app" -> "frontend")
+
+  val frontendDeployment = Deployment(
+    "frontend",
+    DeploymentArgs(
+      spec = DeploymentSpecArgs(
+        selector = LabelSelectorArgs(matchLabels = frontendLabels),
+        replicas = 3,
+        template = PodTemplateSpecArgs(
+          metadata = ObjectMetaArgs(labels = frontendLabels),
+          spec = PodSpecArgs(containers =
+            List(
+              ContainerArgs(
+                name = "frontend",
+                image = "pulumi/guestbook-php-redis",
+                resources = ResourceRequirementsArgs(
+                  requests = Map(
+                    "cpu" -> "100m",
+                    "memory" -> "100Mi"
+                  )
+                ),
+                // If your cluster config does not include a dns service, then to instead access an environment
+                // variable to find the master service's host, change `value: "dns"` to read `value: "env"`.
+                env = List(
+                  EnvVarArgs("name", "GET_HOSTS_FROM"),
+                  EnvVarArgs("value", "dns") /*, EnvVarArgs("value", "env")*/
+                ),
+                ports = List(ContainerPortArgs(containerPort = 80))
+              )
+            )
+          )
+        )
+      )
+    )
+  )
+
+  val frontendService = Service(
+    "frontend",
+    ServiceArgs(
+      metadata = ObjectMetaArgs(labels = frontendLabels, name = "frontend"),
+      spec = ServiceSpecArgs(
+        `type` = isMinikube.map(if _ then ServiceSpecType.ClusterIP else ServiceSpecType.LoadBalancer),
+        ports = List(ServicePortArgs(port = 80, targetPort = 80)),
```

Comment:
the `targetPort` and `containerPort` are logically refering to the same phisical port number, so it would be nice if we create a variable that makes sure they stay in-sync after any changes